packages/mattermost-redux/src/selectors/entities/channels.ts

```diff
@@ -1317,3 +1317,59 @@ export function getDirectTeammate(state: GlobalState, channelId: string): UserPr
 
     return undefined;
 }
+
+export const getRecentProfilesFromDMs: (state: GlobalState) => UserProfile[] = createSelector(
+    'getRecentProfilesFromDMs',
+    getAllChannels,
+    getUsers,
+    getCurrentUser,
+    getMyChannelMemberships,
+    (allChannels: IDMappedObjects<Channel>, users: IDMappedObjects<UserProfile>, currentUser: UserProfile, memberships: RelationOneToOne<Channel, ChannelMembership>) => {
+        if (!allChannels || !users) {
+            return [];
+        }
+        const recentChannelIds = Object.entries(memberships).sort(([, aMembership], [, bMembership]) => {
+            return bMembership.last_viewed_at - aMembership.last_viewed_at;
+        }).map(([cId]) => cId);
+        const groupChannels = Object.values(allChannels).filter((channel: Channel) => channel.type === General.GM_CHANNEL);
+        const dmChannels = Object.values(allChannels).filter((channel: Channel) => channel.type === General.DM_CHANNEL);
+
+        const userProfilesByChannel: {[key: string]: UserProfile[]} = {};
+
+        dmChannels.forEach((channel) => {
+            if (channel.name) {
+                const members = channel.name.split('__');
```

Comment:
You can do this a bit easier by using `getUserIdFromChannelName` from `mattermost-redux/utils/channel_utils` to get the ID of the other user and then getting their profile with `users[id]`